test/index.js

```diff
@@ -18,198 +22,260 @@ function theFuture() {
 }
 
 describe('superagent-auth', function() {
-	beforeEach(function() {
```

Comment:
Sorry this looks like a bigger change than it was. I put all the existing tests inside a "use" category, and added a "tryGetOrigin" category. I also added "adds csrf token for non-relative known URLs" and "sends refreshcookie preflight on boot for known URLs".